File: movieApp/app/src/main/java/com/lbc/practice/movieapp/adapter/PlayingListViewHolder.java
package com.lbc.practice.movieapp.adapter;

import android.os.Handler;
import android.os.Looper;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.lbc.practice.movieapp.callback.OnItemClickListener;
import com.lbc.practice.movieapp.model.resource.local.AppDatabase;
import com.lbc.practice.movieapp.model.Memo;
import com.lbc.practice.movieapp.model.PlayingMovieResult;
import com.lbc.practice.movieapp.R;
import com.lbc.practice.movieapp.callback.Refresh;

import java.util.Date;

/**
 * Created by lbc on 2018-04-02.
 */

public class PlayingListViewHolder extends RecyclerView.ViewHolder {

    ImageView ivImage;
    TextView tvTitle, tvDate, tvScore;
    CheckBox checkBox;
    private AppDatabase appDatabase;
    Refresh refresh;


    public PlayingListViewHolder(View v, final OnItemClickListener callback) {
        super(v);
        v.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (callback != null) {
                    int position = getAdapterPosition();
                    callback.onItemClick(position);
                }
            }
        });
        ivImage = (ImageView) v.findViewById(R.id.iv_item);
        tvTitle = (TextView) v.findViewById(R.id.tv_title);
        tvDate = (TextView) v.findViewById(R.id.tv_date);
        tvScore = (TextView) v.findViewById(R.id.tv_average_score);
        checkBox = (CheckBox) v.findViewById(R.id.checkBox);
        appDatabase = AppDatabase.getsInstance(v.getContext());
    }

    public PlayingListViewHolder(View v, final OnItemClickListener callback, Refresh refresh) {
        super(v);
        v.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (callback != null) {
                    int position = getAdapterPosition();
                    callback.onItemClick(position);
                }
            }
        });
        ivImage = (ImageView) v.findViewById(R.id.iv_item);
        tvTitle = (TextView) v.findViewById(R.id.tv_title);
        tvDate = (TextView) v.findViewById(R.id.tv_date);
        tvScore = (TextView) v.findViewById(R.id.tv_average_score);
        checkBox = (CheckBox) v.findViewById(R.id.checkBox);
        appDatabase = AppDatabase.getsInstance(v.getContext());
        this.refresh = refresh;
    }


    public void setData(final PlayingMovieResult.ResultsBean item) {
        Glide.with(itemView.getContext()).load("http://image.tmdb.org/t/p/w185" + item.getPoster_path()).fitCenter().into(ivImage);


        if (item.getId() == appDatabase.memoDao().judgeMovie(item.getId())) {
            checkBox.setChecked(true);
            checkBox.setText("삭제");

        } else {
            checkBox.setChecked(false);
            checkBox.setText("저장");
        }

        new Handler(Looper.getMainLooper()).postDelayed(new Runnable() {
            @Override
            public void run() {
                checkBox.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {

                        if (checkBox.isChecked()) {
                            checkBox.setText("삭제");
                            appDatabase.memoDao().insertMovie(new Memo(item.getPoster_path(), item.getVote_count(), item.getId(), item.getVote_average(), item.getTitle(),
                                    item.getPopularity(), item.getOverview(), item.getRelease_date(), new Date()));

                        } else {
                            checkBox.setText("저장");
                            appDatabase.memoDao().deleteMovie(appDatabase.memoDao().findMovie(item.getId()));
                            if (refresh != null) {
                                refresh.refresh();
                            }
                        }

                    }
                });

            }
        }, 100);


        tvTitle.setText(item.getTitle());
        tvDate.setText(item.getRelease_date());
        tvScore.setText(item.getVote_average() + "");

    }
}

File: movieApp/app/src/main/java/com/lbc/practice/movieapp/adapter/PlayingListViewHolder.kt
package com.lbc.practice.movieapp.adapter;

import android.os.Handler;
import android.os.Looper;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.lbc.practice.movieapp.R;
import com.lbc.practice.movieapp.callback.OnItemClickListener;
import com.lbc.practice.movieapp.callback.Refresh;
import com.lbc.practice.movieapp.data.Memo;
import com.lbc.practice.movieapp.data.PlayingMovieResult;
import com.lbc.practice.movieapp.data.resource.local.AppDatabase;


import java.util.Date;

/**
 * Created by lbc on 2018-04-02.
 */


class PlayingListViewHolder : RecyclerView.ViewHolder {

    var ivImage: ImageView
    var tvTitle: TextView
    var tvDate: TextView
    var tvScore: TextView
    var checkBox: CheckBox
    private var appDatabase: AppDatabase
    var refresh: Refresh? = null


    constructor(v: View, callback: OnItemClickListener?) : super(v) {
        v.setOnClickListener {
            if (callback != null) {
                val position = adapterPosition
                callback.onItemClick(position)
            }
        }
        ivImage = v.findViewById<View>(R.id.iv_item) as ImageView
        tvTitle = v.findViewById<View>(R.id.tv_title) as TextView
        tvDate = v.findViewById<View>(R.id.tv_date) as TextView
        tvScore = v.findViewById<View>(R.id.tv_average_score) as TextView
        checkBox = v.findViewById<View>(R.id.checkBox) as CheckBox
        appDatabase = AppDatabase.getsInstance(v.context)
    }

    constructor(v: View, callback: OnItemClickListener?, refresh: Refresh) : super(v) {
        v.setOnClickListener {
            if (callback != null) {
                val position = adapterPosition
                callback.onItemClick(position)
            }
        }
        ivImage = v.findViewById<View>(R.id.iv_item) as ImageView
        tvTitle = v.findViewById<View>(R.id.tv_title) as TextView
        tvDate = v.findViewById<View>(R.id.tv_date) as TextView
        tvScore = v.findViewById<View>(R.id.tv_average_score) as TextView
        checkBox = v.findViewById<View>(R.id.checkBox) as CheckBox
        appDatabase = AppDatabase.getsInstance(v.context)
        this.refresh = refresh
    }


    fun setData(item: PlayingMovieResult.ResultsBean) {
        Glide.with(itemView.context).load<Any>("http://image.tmdb.org/t/p/w185" + item.poster_path).fitCenter().into(ivImage)


        if (item.id == appDatabase!!.memoDao().judgeMovie(item.id)) {
            checkBox.isChecked = true
            checkBox.text = "삭제"

        } else {
            checkBox.isChecked = false
            checkBox.text = "저장"
        }

        Handler(Looper.getMainLooper()).postDelayed({
            checkBox.setOnClickListener {
                if (checkBox.isChecked) {
                    checkBox.text = "삭제"
                    appDatabase!!.memoDao().insertMovie(Memo(item.poster_path!!, item.vote_count, item.id, item.vote_average, item.title!!,
                            item.popularity, item.overview!!, item.release_date!!, Date()))

                } else {
                    checkBox.text = "저장"
                    appDatabase!!.memoDao().deleteMovie(appDatabase!!.memoDao().findMovie(item.id))
                    if (refresh != null) {
                        refresh!!.refresh()
                    }
                }
            }
        }, 100)

        tvTitle.text = item.title
        tvDate.text = item.release_date
        tvScore.text = item.vote_average.toString() + ""

    }
}


